docs/vcpkg-schema-definitions.schema.json

```
@@ -800,18 +799,18 @@
         }
       ]
     },
-    "port-name": {
+    "package-pattern": {
       "type": "string",
-      "description": "Name of a package.",
+      "description": "A pattern to match package name.",
       "allOf": [
         {
-          "description": "Package name must be a dot-separated list of valid identifiers",
-          "pattern": "^[a-z0-9]+(-[a-z0-9]+)*(\\.[a-z0-9]+(-[a-z0-9]+)*)*$"
+          "$comment": "https://learn.microsoft.com/en-us/vcpkg/reference/vcpkg-configuration-json#registry-packages",
+          "description": "Package pattern may contain only lowercase letters, digits, and -, with an optional trailing *",
+          "pattern": "^[a-z0-9]+(-[a-z0-9]+)*(-?\\*)*$"
```

```suggestion
          "pattern": "^[a-z0-9]+(-[a-z0-9]+)*-?\\*$"
```

`abc------****` isn't valid but would be as written because `-?\\*` was in a `*`

---

```suggestion
          "pattern": "^[a-z0-9]+(-[a-z0-9]+)*(-?\\*)?$"
```
unless you `really mean abc-def-*-*-*`.

And
```suggestion
          "pattern": "^[a-z0-9]+(-[a-z0-9]+)*\\*?$"
```
if the `*` may follow letters.

And
```suggestion
          "pattern": "^[a-z0-9]+(-[a-z0-9]+)*[*]?$"
```
if you dislike backslashes...

---

Fixed.
* Change quantifier `*` with `?` to match one group at most.
* Change character `\*` with `[*]` to reduce escaping